### ISSUE TITLE
[Feature request] Migrate jQuery to 4.0 Beta and upcoming final release

### ISSUE THREAD

**Issue by gebederry** [jeanropke/RDR2CollectorsMap#1242]
Hello,
JQuery 4.0 beta has been released for some time now, im looking forward to the performance improvements mentioned in announcement, im sure you do.
Thank you for the long-term maintaining!

**Comment by MichalD96** [comment]
Hello.
We will think about updating if JQuery 4.0 is in stable release.
New version very likely will not be backward compatible.
If update will require hundreds of changes in the code, probably is not worth it.
Most crucial performance operations are in vanilla JavaScript anyway.
Where do you expect performance improvement after the update?
If we look for performance improvement in specific part of the code, would be the best to use vanilla JS.

**Comment by gebederry** [comment]
> Hello. We will think about updating if JQuery 4.0 is in stable release. New version very likely will not be backward compatible. If update will require hundreds of changes in the code, probably is not worth it. Most crucial performance operations are in vanilla JavaScript anyway. Where do you expect performance improvement after the update? If we look for performance improvement in specific part of the code, would be the best to use vanilla JS.

I can't tell the specific contents that need to be improved technically, I feel like another map site [RDR2Map](https://rdr2map.com/) in modern framework has smoother experience. Perhaps the new version will bring a little improvements. And I agree with you its not worth if really just _a little_ XD

**Comment by MichalD96** [comment]
RDR2Map uses [Mapbox](https://www.mapbox.com/), which is something totally different to what we use. Maybe is better optimized than leaflet.js
Something that would definitely improve our map performance, is to split each item category on new map layer. It was like that on the beginning, i don't really remember why somebody changed that. You can see that difference on [RDOMap](https://jeanropke.github.io/RDOMap/). Way more markers, and much smoother map dragging.

**Comment by gebederry** [comment]
> RDR2Map uses [Mapbox](https://www.mapbox.com/), which is something totally different to what we use. Maybe is better optimized than leaflet.js Something that would definitely improve our map performance, is to split each item category on new map layer. It was like that on the beginning, i don't really remember why somebody changed that. You can see that difference on [RDOMap](https://jeanropke.github.io/RDOMap/). Way more markers, and much smoother map dragging.

Sorry for not replying for so long, I thought there was anything substantial to add or to occupy a reply in the issue. 
Indeed that is pain in the ass, even just toggling 'fast travel' category will cause all of item markers re-rendering or, in other words, removing entire layer and `addMarkers()` re-adding it again. I was trying to fix this problem but I found I couldn't do it :/ 
Hope you or someone else can rewrite this section and separate each category into independent layers.
https://github.com/jeanropke/RDR2CollectorsMap/blob/3193341a7cecccdc7c9a8dec03596d1c312946a8/assets/js/menu.js#L129-L144
It would be uplifting if this major performance issue got fixed, although this is a legacy repo. New version or framework... isn't really necessary anymore.
